### PR TITLE
clients: add shisui  (golang) portal client 

### PR DESCRIPTION
**Pull request by fearlessfe**
add new portal client shisui  #1019

### CODE REVIEW

File: clients/shisui/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+ARG branch=latest
+
+FROM ghcr.io/optimism-java/shisui:$branch
```
Comment (KolbyML) [inline clients/shisui/Dockerfile:3]:
```suggestion
ARG baseimage=ghcr.io/optimism-java/shisui
ARG tag=latest

FROM $baseimage:$tag
```

File: clients/shisui/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+RUN echo "latest" > /version.txt
+
+# Export the usual networking ports to allow outside access to the node
+EXPOSE 8545 9009/udp
+
+ENTRYPOINT ["app"]
```
Comment (KolbyML) [inline clients/shisui/Dockerfile:10]: add a new line to the end of this file

File: clients/shisui/hive.yaml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+roles:
+  - portal
```
Comment (KolbyML) [inline clients/shisui/hive.yaml:2]: add a new line to the end of this file

File: clients/shisui/Dockerfile (hunk 2 above)
Comment (fearlessfe) [reply clients/shisui/Dockerfile:10]: ok

**Comment by GrapeBaBa** [comment]
@fearlessfe It would be better to create an issue associated with this PR.

File: clients/shisui/Dockerfile (hunk 1 above)
Comment (KolbyML) [reply clients/shisui/Dockerfile:3]: Could you make this change as well? As this format is used to pass which tag you want to use when running hive tests

File: clients/shisui/hive.yaml (hunk 3 above)
Comment (KolbyML) [reply clients/shisui/hive.yaml]: This file still indicting an invalid page ending

File: clients/shisui/Dockerfile (hunk 1 above)
Comment (fearlessfe) [reply clients/shisui/Dockerfile]: Got it

File: clients/shisui/hive.yaml (hunk 3 above)
Comment (KolbyML) [reply clients/shisui/hive.yaml]: image
by that I mean this red circle. Once it is gone I am good to merge this

**Comment by KolbyML** [comment]
@fearlessfe I made a PR on your fork with the fix https://github.com/optimism-java/hive/pull/1

**Pull request merged**